v1-2/learn/how-to-use-java-interoperability.md

```diff
@@ -0,0 +1,939 @@
+   class: "java.util.ArrayDeque"
+} external; 
+```
+
+If you compare these functions with the Java method signatures, you would notice the additional parameter `handle arrayDequeObj` in Ballerina functions. Let’s look at a sample usage to understand the reason. 
+
+```ballerina
+public function main() {
+   // Create a new instance of ArrayDeque
+   handle arrayDequeObj = newArrayDeque();
+
+   // Convert a Ballerina string to a Java string 
+   string str = “Ballerina”
+   handle handleStr = java:fromString(str);
+
+   push(arrayDequeObj, handleStr);
+   handle e = pop(arrayDequeObj);
+}
+```
+
+As you can see, you need to first construct an instance of the `ArrayDeque` class. The variable `arrayDequeObj` refers to an `ArrayDeque` object. Then, you need to pass this variable to both `pop` and `push` functions because the corresponding Java methods are instance methods of the class `ArrayDeque`. Therefore, you need an instance of the `ArrayDeque` class in order to invoke its instance methods. You can think of the variable `arrayDequeObj` as the method receiver.
+
+#### Map Java classes into Ballerina objects
+The following pattern is useful If you want to present a clearer Ballerina API that calls to the underneath Java code. This pattern creates wrapper Ballerina objects for each Java class that you want to expose via your API. 
+
+Imagine that you want to design an API to manipulate a stack of string values by using the Java `ArrayDeque` utility. You can create a Ballerina object type as follows. 
+
+```ballerina
+public type StringStack object {
+   private handle jObj;
+
+   public function __init() {
+       self.jObj = newArrayDeque();
+   }
+
+   public function push(string element) {
+       push(self.jObj, java:fromString(element));
+   }
+
+   public function pop() returns string {
+       handle handleEle = pop(self.jObj);
+       // Let's talk about error handling and null satefy later in this guide
+       // This example uses an empty string for now.
+       return java:toString(handleEle) ?: "";
+
+   }
+};
+
+function newArrayDeque() returns handle = @java:Constructor {
+   class: "java.util.ArrayDeque"
+} external;
+
+function pop(handle receiver) returns handle = @java:Method {
+   class: "java.util.ArrayDeque"
+} external;
+
+function push(handle receiver, handle element) = @java:Method {
+   class: "java.util.ArrayDeque"
+} external;
+```
+
+This object presents a much clearer API compared to the previous API. Here is a sample usage of this object. 
+
+```ballerina
+public function main() {
+   StringStack stack = new();
+   stack.push("Ballerina");   
+   string element = stack.pop();
+}
+```
+
+#### Overloaded Java methods
+The“Instantiate Java classes” section presented about how to deal with overloaded constructors in the. You need to use the same approach to deal with overloaded Java methods. Let’s try to call the overloaded `append` methods in the `java.lang.StringBuffer class Here is a subset of those methods. 
```

Comment:
```suggestion
The “Instantiate Java classes” section presented about how to deal with overloaded constructors in the. You need to use the same approach to deal with overloaded Java methods. Let’s try to call the overloaded `append` methods in the `java.lang.StringBuffer class. Here, is a subset of those methods. 
```
Better to link to the “Instantiate Java classes” section. That sentence is incomplete.